lib/src/https_openssl.rs

```diff
@@ -597,13 +596,13 @@ impl Session {
             });
 
             self.backend.as_ref().map(|backend| {
```

Comment:
To keep consistency with the http.rs implementation

Use a `if let Some(x)` instead of `map`

```suggestion
if let Some(ref backend) = self.backend {

            let backend = &mut *backend.borrow_mut();
            backend.failures += 1;

            let already_unavailable = backend.retry_policy.is_down();
            backend.retry_policy.fail();
            incr!(
                "connections.error",
                self.cluster_id.as_deref(),
                self.metrics.backend_id.as_deref()
            );

            if !already_unavailable && backend.retry_policy.is_down() {
                error!(
                    "backend server {} at {} is down",
                    backend.backend_id, backend.address
                );
                incr!(
                    "down",
                    self.cluster_id.as_deref(),
                    self.metrics.backend_id.as_deref()
                );

                push_event(ProxyEvent::BackendDown(
                    backend.backend_id.clone(),
                    backend.address,
                ));
            }

        }
```

Reply:
Thanks :)